includes/discovery/functions.inc.php

```diff
@@ -135,7 +135,14 @@ function discover_device($device, $options = null)
         }
     }
     foreach ($config['discovery_modules'] as $module => $module_status) {
-        if ($force_module === true || $attribs['discover_' . $module] || ( $module_status && !isset($attribs['discover_' . $module]))) {
+        $os_module_status = $config['os'][$device['os']]['discovery_modules'][$module];
+        d_echo("Global module enabled: ".$module_status. "\n");
+        d_echo("OS module enabled    : ".$os_module_status. "\n");
+        d_echo("Device module enabled: ". $attribs['discover_' . $module]. "\n");
+        if ($force_module === true ||
+            $attribs['discover_' . $module] ||
+            ($os_module_status && !isset($attribs['discover_' . $module])) ||
+            ($module_status && !isset($os_module_status) && !isset($attribs['discover_' . $module]))) {
```

Comment:
`isset($os_module_status)` won't work as expected.  You'll need to do `isset($config['os'][$device['os']]['discovery_modules'][$module])`